### PR TITLE
setup work

### PR DESCRIPTION
**Pull request by Isaac-Lozano**
This one includes a more complex `.travis.yml`. `Cargo.toml` has been changed to work off of master branch.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+rust:
+  - stable
+  - beta
+  - nightly
+os:
+  - windows
+  - linux
+  - osx
+before_deploy:
+  - if [[ "$TRAVIS_OS_NAME" == "windows" ]]; then mv target/release/backupd.exe backupd_windows-x64.exe; fi
```
Comment (lietu) [inline .travis.yml]: Maybe grouping these `== "windows"` bits either so the lines are below each other, or putting the two commands in the same `if` would be clearer - what do you think?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then DEPLOY_FILE="backupd_linux-x64"; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then DEPLOY_FILE="backupd_osx-x64"; fi
+matrix:
+  allow_failures:
+    - rust: nightly
+  fast_finish: true
+deploy:
+  provider: releases
+  api_key:
+    secure: cQnYkESvIJiQHzWwYFf7/s8m/qCYc/4JhcgtHTPl72yop9TR4AdVLVPD+0J0DUpL/cihxz+AoI57uLCSjbjIcozAuA2dTRGcaBUbdxx70kqNwbPX2Gs6YYswWH2gY6sj3ptCfJTe99C0LkGBumuf9izrrfYO7Z3QptnCfOCUjn4ZHvvDbUlPx/XDBCbLSZmOekHvD0OG758JcPyWtZx2dkOmmU5piyzusU4aVUzFqMZrggPEjj2ZNqn1GpWtrja1HpwpFTjYlQ/JS2FZMN1VPncvRGdNQVLSKXTv+qGVYYfNLHdv5sOa2LvcXbfrmICS4sN8NtyVcM31OLyzgz7B9d3rFhqFtXCJrfyTgwcBqMZXIu+/o90lSYyCOzUX4TIf8w3IQJLDP+lFr3B4lIuwDLPEm8sleIQPUgGJkFHOGiTJXrbci6VjgrLlrlMCVQgkTOKZRbnVBiZXJq42v+rZbAhyXv+y4VbJachcmrp/uuQDsEMdD/U1BvY1JfbHS8AgzuFMy0xbZp+rgoLAcsCfnWyzGzJ78qa5JHECTvKzAwEs0Eg5VrffgsEHGTrPCv5L7hYntHWprV7U2ParubTBO7YeBrSRczqcwZCDPyZjLrSwYcv0JoB2307VlaKrga6DVOLL57O0nOdJLNh58qK4c3dnkUOiHWQPF8QEsf1NQ7E=
```
Comment (lietu) [inline .travis.yml]: Any reason not to use ENV variables in Travis-CI instead? tbh I don't know of any benefits in either solution

File: Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "backupd"
+version = "0.1.0"
+authors = ["Isaac Lozano <109lozanoi@gmail.com>"]
+edition = "2018"
+
+[dependencies]
+
+[dependencies.backuplib]
+git = "https://github.com/Lieturd/backuplib"
```
Comment (lietu) [inline Cargo.toml]: Do you know if this syntax is preferred over

```
backuplib = { git = "https://github.com/Lieturd/backuplib" }
```

?

The only quick reason for or against either that I can think of, is this one-line alternative is shorter

File: .travis.yml (hunk 1 above)
Comment (Isaac-Lozano) [reply .travis.yml]: I can toss these into a literal so that it looks nicer and doesn't have redundant checks.

File: .travis.yml (hunk 2 above)
Comment (Isaac-Lozano) [reply .travis.yml]: I don't think env variables can work in this situation. I would have to see what you mean, but this seems to be the standard way to do this. The key is encrypted in such a way that only travis can read it.

File: Cargo.toml (hunk 3 above)
Comment (Isaac-Lozano) [reply Cargo.toml]: I normally use the one-line notation. This more explicit version was a byproduct of when I had `branch="setup"`. I don't expect we'd need to do that much ever again, so I can switch to the shorter version.

File: .travis.yml (hunk 2 above)
Comment (lietu) [reply .travis.yml]: https://docs.travis-ci.com/user/environment-variables/#defining-variables-in-repository-settings <- this is what I mean, you define them in their system. It's similarly write-only for us, but it's not visible even in encrypted format in the repository.

Minor difference, I guess just seeing `$DEPLOY_KEY` in the config makes more sense to me than an encrypted value.

File: Cargo.toml (hunk 3 above)
Comment (lietu) [reply Cargo.toml]: Does the one-liner not support having also `branch` defined via e.g. `git = "", branch = ""`?

**Pull request merged**